addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -795,10 +865,43 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
      * @override
      */
     onFocus: function () {
+        this._updateAvailableVisibilityDependencies();
+        const dependencyEl = this._getDependency();
+        if (dependencyEl && (['radio', 'checkbox'].includes(dependencyEl.type) || dependencyEl.nodeName === 'SELECT')) {
+            this._updateAvailableVisibilityOptions();
+            return;
+        }
         // Other fields type might have change to an existing type.
         // We need to reload the existing type list.
         this.rerender = true;
     },
+    /**
+     * Checks if the deleted field was the visibility dependency of others fields, if yes makes them visible
+     * 
+     * @override
+     */
+    onRemove() {
+        const nameOfDeletedElement = this.$target[0].querySelector('.s_website_form_input').name;
+        const dependentFieldContainerEl = document.querySelectorAll(`[data-visibility-dependency="${nameOfDeletedElement}"]`);
+        for (const fieldContainerEl of dependentFieldContainerEl) {
+            this._deleteConditionalVisibility(fieldContainerEl);
+        }
+    },
+    /**
+     * Synchronizes the fields that have the same name as the field that is losing focus
+     * 
+     * @override
+     */
+    onBlur() {
+        const currentFieldEl = this.$target[0];
+        const nameOfCurrentField = currentFieldEl.querySelector('.s_website_form_input').name;
+        for (const fieldEl of document.querySelectorAll('.s_website_form_field')) {
+            const inputsWithSameName = fieldEl.querySelectorAll(`.s_website_form_input[name="${nameOfCurrentField}"]`);
+            if (inputsWithSameName.length > 0 && !currentFieldEl.isSameNode(fieldEl)) {
+                this._synchronizeFields(currentFieldEl, fieldEl);
+            }
+        }
+    },
```

Comment:
Instead of synchronizing each element individually, I opted to synchronize everything every time the element loses focus.
It's easier than having to

- Synchronize the default values

- Synchronize the different possible values for select, checkbox and radios

- Synchronize the placeholders,

- ...

Reply:
I don't see why this is easier :/ This is kinda the same debate as `cleanForSave` instead of directly by the option method: an option is changed, I want the DOM to be consistent directly just after the option method has done its job. Using `onBlur` to make editions is actually probably buggy as soon as you use undo/redo: my guess is that there will be two undo records: one for the user edition and one for the synchronization... and there could have been other edition in between. Actually I just tried and it is indeed buggy: create two fields with the same name -> set "hello" as default value, add a description (so "hello" is not yet synchronized on the second input), blur (synchronized) -> CTRL-Z de-synchronizes the two values -> your form is now broken.

Also I don't think you should synchronize the placeholders for example, you could want different placeholder according to the condition you make the input visible. For me the only thing to synchronize is default value/ possible values and type -> so this synchronization can be done in those respective areas. Example for default value: the option asks the form option "Each option with XXX name now use YYY as default value".